bokeh/models/widgets/sliders.py

```diff
@@ -63,6 +67,11 @@ def __init__(self, **kwargs):
         if 'start' in kwargs and 'end' in kwargs:
             if kwargs['start'] == kwargs['end']:
                 raise ValueError("Slider 'start' and 'end' cannot be equal.")
+
+        if 'format' in kwargs:
+            if isinstance(kwargs["format"], str):
+                warnings.warn("Passing 'format' string will be deprecated in future releases. Use TickFormatter derived classes instead.")
+
```

Comment:
I don't know that we would consider this deprecated. Even if we eventually change the property type to be `Instance(TickFormatter)` we might add an `.accepts(String, ...)` clause that converts a string in to an appropriate formatter as a convenience. Id' suggest holding off on declaring this deprecated just now (we can always add the deprecation any time throughout 2.x) if we change mind.